Documentation edits in preparation for alpha release (#691)
swarm-game/swarm: README.md

      without worrying about collecting resources.
    - There are also challenge scenarios where you attempt to program
      robots in order to solve pre-designed puzzles or challenges.

Installing
==========

> Instructions for installing from a binary release will go here!

Reviewer: ```suggestion
There will _soon_ be a release of Swarm on Hackage and in a binary form!

If you are interested in Haskell or functional programming, you can install
it from source (see below) or once it is released with the Cabal tool.

For instructions on how to install the binary directly check the releases.
```

Author: Oops, missed this comment too.  Will fix.

Installing from Source

> ======================

Reviewer: ```suggestion
----------------------
```

If you want to contribute, you're most welcome!  There are *lots* of
ways to contribute, regardless of your Haskell background.  For
example, even someone with no Haskell experience whatsoever could
still help with *e.g.* game design, playtesting, and creating

> challenge scenarios.  Check out the [CONTRIBUTING](CONTRIBUTING.md)

Reviewer: ```suggestion
challenges and scenarios.  Check out the [CONTRIBUTING](CONTRIBUTING.md)
```

Author: I mean, challenges are a type of scenario... but people reading this document don't know that.  So I guess this is better. :smile:
